src/posix.c

```diff
@@ -235,7 +235,7 @@ int git__mmap_alignment(size_t *alignment)
 }
 
 
-int p_mmap(git_map *out, size_t len, int prot, int flags, int fd, git_off_t offset)
+int p_mmap(git_map *out, size_t len, int prot, int flags, int fd, off64_t offset)
```

Comment:
Quick question, no need to change: why did you opt to convert to `off64_t` instead of just typedeffing `git_off_t` to `off64_t`?

Reply:
I thought that it was clearer for internal uses to use something Very Obviously 64 bit, and something that is a Standard Type.  `git_off_t` feels like the sort of thing that is perhaps useful for our external API, but less useful for our internal.

Your comment did make me rethink a few things, so I actually kept `git_off_t` in the external type places that it was used.  I think this is a pretty reasonable compromise.